### PR TITLE
[REVIEW] Enable `fixed_point` binary operations

### PR DESCRIPTION
**Pull request by codereport**
This addresses are part of https://github.com/rapidsai/cudf/issues/3556. It is a follow up to https://github.com/rapidsai/cudf/pull/5861 where most of the work was done. However, this PR fixes an issue with `binary_operator::DIV`.

Starting point is to comment in the commented out unit tests in `binop-integration-tests.cpp` and go from there.

### Update:

After https://github.com/rapidsai/cudf/pull/6544 got merged, this job was simplified to the following:

* [x] Enable `bool is_supported_fixed_point_binop(binary_operator op)`
* [x] Uncomment out tests
* [x] Modify `FixedPointBinaryOpDiv` test
* [x] Add a couple more `binary_operator::DIV` tests

### CODE REVIEW

**Comment by GPUtester** [comment]
Please update the changelog in order to start CI tests.

View the gpuCI docs [here](https://docs.rapids.ai/gpuci#open-pr-or-update-pr).

**Comment by harrism** [comment]
Please target branch-0.17.

**Comment by sperlingxx** [comment]
For current implementation,  it seems that fixed_point types are not able to perform binary opeartion with other numeric types. Will we enable it in this ongoing PR?

**Comment by codereport** [comment]
> For current implementation, it seems that fixed_point types are not able to perform binary opeartion with other numeric types. Will we enable it in this ongoing PR?

No, that is not a part of this PR.

**Comment by harrism** [comment]
@sperlingxx Once (unary) casts are implemented the approach would be to cast to fixed point and then do the binary operation.

**Comment by sperlingxx** [comment]
> @sperlingxx Once (unary) casts are implemented the approach would be to cast to fixed point and then do the binary operation.

I got it. Thanks, Mark!

**Review by devavret — CHANGES_REQUESTED** [review]
This is from a previous PR but during the creation of temporary result in `fixed_point_binary_operation`, `cudf::binary_operation` is being used to create a temporary column of the same scale. This does not take a stream. Try using the `detail` version.

File: cpp/src/binaryop/binaryop.cpp [hunk 1]
```diff
@@ -371,8 +371,7 @@ bool is_comparison_binop(binary_operator op)
  */
 bool is_supported_fixed_point_binop(binary_operator op)
 {
-  // TODO in a follow up PR
-  return false;  // is_basic_arithmetic_binop(op) or is_comparison_binop(op);
+  return is_basic_arithmetic_binop(op) or is_comparison_binop(op);
```
[the remark this answers is not in the document]
Comment (codereport) [reply cpp/src/binaryop/binaryop.cpp:374]: You are right. That is an oversight on my part. I will add to https://github.com/rapidsai/cudf/pull/6544 as it is getting merged sooner.

**Comment by codereport** [comment]
This job is currently waiting on https://github.com/rapidsai/cudf/pull/6544 to be merged.

**Review by trxcllnt — APPROVED** [review]

**Review by harrism — APPROVED** [review]

**Review by devavret — APPROVED** [review]

**Pull request merged**